app/api/images/images.controller.js

```diff
@@ -456,6 +456,50 @@ exports.getAttributes = utils.route(async (req, res) => {
   });
 });
 
+// GET /images/:id/geopose
+// =========================
+
+// Convenience function for computing the GeoPose of an image
+function mod(n, m) {
+  return ((n % m) + m) % m;
+}
+
+exports.getGeoPose = utils.route(async (req, res) => {
+  const queryPromise = models.images.findOne({
+    attributes: [
+      "id",
+      "state",
+      "azimuth",
+      "tilt",
+      "roll",
+      [models.sequelize.literal("ST_X(location)"), "lon"],
+      [models.sequelize.literal("ST_Y(location)"), "lat"],
+      [models.sequelize.literal("ST_Z(location)"), "h"],
+    ],
+    where: {
+      id: req.params.id,
+      state: {
+        [Op.or]: ['validated','waiting_validation']
+      }
+    }
+  });
+  const query = await queryPromise;
+  const result = query.toJSON();
```

Comment:
could you check how it behave for non georef image and ideally return 404 in such cases?

Reply:
Commit https://github.com/MediaComem/smapshot-api/pull/71/commits/8d87e0d5485854cdf3f6f8639d65d3e55117b1f7 should resolve this.